### PR TITLE
Group Support - Basic CRUD

### PR DESCRIPTION
**Pull request by dilannery**

### CODE REVIEW

File: lib/user.js [hunk 1]
```diff
@@ -105,5 +105,28 @@ module.exports = {
...
   denyTracking: function(){
     refs.userMeta.child('trackOk').set( false );
+  },
+
+  /*
+   * Group Support
+   */
+  setGroup: function(key, devices) {
+    if (devices === null) { // Just create group without any device.
+      refs.userGroups.child(key).set(false);
```
Comment (eighteyes) [inline lib/user.js]: throw a log here 'Empty group created : {key}'

File: lib/user.js [hunk 2]
```diff
@@ -105,5 +105,28 @@ module.exports = {
...
+    }
+    refs.userGroups.child(key).set(devices);
+  },
+  deleteGroup: function(key) {
+    refs.userGroups.child(key).set(null);
```
Comment (eighteyes) [inline lib/user.js]: why not `remove()` ?

**Review by eighteyes — APPROVED** [review]
looks good, please fix comments

File: lib/user.js (hunk 2 above)
Comment (dilannery) [reply lib/user.js]: removeGroup()? Just remove() isn't semantic because it'll be inside `matrix-firebase.user.[remove]()`

**Pull request merged**